pages/register.tsx

```diff
@@ -219,3 +220,5 @@ export default function Register() {
     </Page>
   )
 }
+
+export default flags.registration ? Register : withComingSoon(Register)
```

Comment:
maybe a slight variation to this pattern:
What if you always returned the "wrapped" version of the component and by also passing in the flag it allows the ternary to be in the single wrapper HOC. So something like:
```js
export default withDelayedRelease(Register, flags.registration)
```

Reply:
or maybe even add a "date" capability like: 
```js
export default withReleaseOn(Register, flags.registrationDate)
```
Where the HOC checks if today's date is on/after the configured date. I'm not sure how Next handles this with regards to built-time vs runtime. So we'd need to make sure the flags are computed via api or at least dynamically re-evaluated on load. It would be unfortunate if someone have the app open the day before and navigated around and still saw coming soon!